### PR TITLE
Add tests for error handling behaviour of GetAll and Mirror

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
Blocked on https://github.com/sociomantic-tsunami/swarm/milestone/25.

**Comment by gavin-norman-sociomantic** [comment]
Updated to swarm v4.4.0. Unblocked.

**Comment by gavin-norman-sociomantic** [comment]
Added a reconnection test for Mirror.

**Comment by gavin-norman-sociomantic** [comment]
Damn, segfault while running the test. Trying to reproduce locally. (No luck so far.)

**Comment by gavin-norman-sociomantic** [comment]
Hmm... And now all my retests in Travis succeed :/

**Comment by gavin-norman-sociomantic** [comment]
Well... after about 10 retests (each running the test suite 4 times), the segfault hasn't returned. I guess there's an obscure bug in there somewhere, but I can't think of a way to find it right now. It's also likely that the bug has existed for some time and is unrelated to this PR.

I'd suggest we just go ahead with review / merge and hope that the bug either never pops up again or becomes more repeatable at some point.

File: src/dhttest/cases/neo/GetAll.d [hunk 1]
```diff
@@ -164,6 +164,80 @@ public class GetAllSuspend : NeoDhtTestCase
...
+        ubyte[] val;
+        val.length = 8 * 1024;
+        for ( hash_t key = 0; key < num_records; key++ )
+        {
+            foreach ( ref b; val )
+                b = rand.uniform!(ubyte)();
+
+            auto res = this.dht.blocking.put(this.test_channel, key, val);
+            test(res.succeeded);
+        }
```
Comment (nemanja-boric-sociomantic) [inline src/dhttest/cases/neo/GetAll.d]: Use the function you just added in the previous commit ;-)

Comment (nemanja-boric-sociomantic) [reply src/dhttest/cases/neo/GetAll.d]: `putRecords`. Unless it's GitHub's wrong commit order thing?

Comment (gavin-norman-sociomantic) [reply src/dhttest/cases/neo/GetAll.d]: I think you're reading backwards.

File: src/dhttest/cases/neo/GetAll.d [hunk 2]
```diff
@@ -164,6 +164,80 @@ public class GetAllSuspend : NeoDhtTestCase
...
+
+        uint disconnection_count;
+        auto getall = GetAll(this.dht);
+        getall.start(this.test_channel,
+            ( DhtClient.Neo.GetAll.Notification info,
+                DhtClient.Neo.GetAll.Args args )
+            {
+                with ( info.Active ) switch ( info.active )
+                {
+                    case started:
```
Comment (nemanja-boric-sociomantic) [inline src/dhttest/cases/neo/GetAll.d]: Would it be of help if you dropped connections somewhere in the half of the iteration? Maybe drop connection twice - once here and once when on the 50% of `num_records`?

Comment (gavin-norman-sociomantic) [reply src/dhttest/cases/neo/GetAll.d]: Oh yeah, you're right. That would make more sense. I think that's what I intended ;)

Comment (gavin-norman-sociomantic) [reply src/dhttest/cases/neo/GetAll.d]: Updated. (Tested in dhtnode.)

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**